### ISSUE TITLE
Latent Semantic Analysis

### ISSUE THREAD

**Issue by ermakovpetr** [nltk/nltk#936]
What do you think about adding a Latent Semantic Analysis (LSA) in NLTK?
I am often faced with the need to apply the LSA in my tasks and would like to have it in NLTK.

I am ready to make the implementation of this algorithm in NLTK and would like to hear the opinion of mentors on this issue.

**Comment by stevenbird** [comment]
@ermakovpetr are there implementations in sklearn or numpy that would do the job? Then it might be just a matter of demonstrating how to use it in connection with other NLTK functionality. That would be a welcome contribution. We can easily host examples on the website (e.g. in http://www.nltk.org/howto/).

**Comment by ewan-klein** [comment]
I think it would be great to have support for LSA in NLTK, though I agree with @stevenbird that we ought to reuse as much as possible from existing libraries. There is a tutorial and a simple Python implementation here: http://www.puffinwarellc.com/index.php/news-and-articles/articles/33.html?showall=1

**Comment by kmike** [comment]
https://github.com/piskvorky/gensim is a go-to Python package for topic modelling; it has LSA: https://radimrehurek.com/gensim/models/lsimodel.html. I think pointing NLTK users to gensim is better than creating another implementation in NLTK.

**Comment by stevenbird** [comment]
+1

**Comment by ermakovpetr** [comment]
@kmile, yes, but where is the lane line?
**LSA** is in the **gensim**. After **LSA** I would add **pLSA**, but where? to **gensim** too?
**NLTK** is much more popular than the **gensim**. These algorithms will see and take advantage of more people.

**Comment by kmike** [comment]
> LSA is in the gensim. After LSA I would add pLSA, but where? to gensim too?

Yes, I think gensim is a right place for pLSA, although gensim author said pLSA implementation is not needed - see https://groups.google.com/forum/#!topic/gensim/f6fad21h1Xg

Some advantages of having such code in gensim:
- less code to maintain for NLTK;
- many topic modelling algorithms in the same package;
- they can share common utility code;
- easier to test - unfortunately, NLTK testing suite is heavy, and it doesn't work on Travis CI currently, so no integration with github;
- no community splitting;
- no duplicate work.

While NLTK is well-known and popular, gensim is well-known and popular too, so I don't see a problem here.

**Comment by stevenbird** [comment]
Yes, we should just interface to gensim.

**Comment by ermakovpetr** [comment]
http://scikit-learn.org/stable/modules/generated/sklearn.decomposition.TruncatedSVD.html
